tool/generate-trufflemock.rb

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ruby
+
+#
+# Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved.
+# This code is released under a tri EPL/GPL/LGPL license. You can use it,
+# redistribute it and/or modify it under the terms of the:
+#
+# Eclipse Public License version 1.0
+# GNU General Public License version 2
+# GNU Lesser General Public License version 2.1
+#
+
+require 'erb'
+
+types = { 'void *' => "0", 'void ' => nil, 
+    'bool ' => 'false', 'int ' => '0', 'long ' => '0', 'char ' => "'0'",
+    'float ' => '0.0', 'double ' => '0.0'}
```

Comment:
I think we should avoid the confusing extra space in keys here, and use `match[1].gsub(' ', '')` below.